### DISCUSSION TITLE
ApkSigner error in Flash Professional CC (v13.1.0.226) (Windows 10)

### DISCUSSION THREAD

**Discussion by MalacTheLittle** [airsdk/Adobe-Runtime-Support#1342] [Q&A]
Call me old-fashioned, but I'm so familiar to Flash CC and don't feel nice to upgrade it to Animate if there's no real need for it.
When packaging .apk file with newer versions of AIR SDK (554 or 633), I'm getting error with ApkSigner: 

> Exception in thread "main" java.lang.UnsupportedClassVersionError: com/android/apksig/ApkSigner$SignerConfig$Builder : Unsupported major.minor version 52.0
> 	at java.lang.ClassLoader.defineClass1(Native Method)
> 	at java.lang.ClassLoader.defineClass(Unknown Source)
> 	at java.security.SecureClassLoader.defineClass(Unknown Source)
> 	at java.net.URLClassLoader.defineClass(Unknown Source)
> 	at java.net.URLClassLoader.access$100(Unknown Source)
> 	at java.net.URLClassLoader$1.run(Unknown Source)
> 	at java.net.URLClassLoader$1.run(Unknown Source)
> 	at java.security.AccessController.doPrivileged(Native Method)
> 	at java.net.URLClassLoader.findClass(Unknown Source)
> 	at java.lang.ClassLoader.loadClass(Unknown Source)
> 	at sun.misc.Launcher$AppClassLoader.loadClass(Unknown Source)
> 	at java.lang.ClassLoader.loadClass(Unknown Source)
> 	at com.adobe.air.apk.APKSigner.apkSign(APKSigner.java)
> 	at com.adobe.air.apk.APKOutputStream.d(APKOutputStream.java)
> 	at com.adobe.air.apk.APKPackager.renameOutput(APKPackager.java)
> 	at com.adobe.air.ApplicationPackager.createPackage(ApplicationPackager.java)
> 	at com.adobe.air.ADT.parseArgsAndGo(ADT.java)
> 	at com.adobe.air.ADT.run(ADT.java)
> 	at com.adobe.air.ADT.main(ADT.java)
> 

Using AIR SDK 444 doesn't throw any error.
Altough I may be old-fashioned, I have Animate installed on my computer just for debuging these kind of errors :) So, with newer SDKs there's no error with ApkSigner in Animate. 
Do you know is there any way to fix this issue? What part could be causing this?

Thanks

EDIT:
I have installed jdk1.8.0_144 and in Environmental variables path AIR SDK is set to version 633.

**Comment by ajwfrost** [comment]
The unsupported class version is where a Java runtime is trying to load a Java class file that was built to target a later version of the Java runtime. Version 52 is Java 8, and this is the minimum required for the Android tools (such as apksigner).

So what's most likely happening is that when you're running via Flash Professional, it's using their built-in version of Java and is then unable to load in the later Java-based tools. I'm not sure whether it's possible to update the Java version being used in Flash Professional in the same way that you can update the JRE from Flash Builder.. are you able to see whether your installation has got a JRE built into it, or whether it's set up to use a particular JRE available on your machine?

**Reply by MalacTheLittle** [reply]
Thanks Andrew, once again your hint solved my problem :)

I've located JRE folder that Flash CC uses, on Windows 10 it's `C:\Program Files\Common Files\Adobe\Adobe Flash CC\jre`. I've copied `jre1.8.0_144` folder and renamed it to `jre` and it workes as a charm :)

Just one more question: You mentioned at #1101 the latest version we should use is JRE 11 and not newer. Would there be any benefit to use JRE 11 instead of JRE 8? And which one is the best to use for AIR SDK 31.1.1.633?

Thanks!

**Reply by ajwfrost** [reply]
Currently I think the minimum JRE that you can use is 8 (aka 1.8) in order to work with some of the Android tools: but, with the latest Android build tools and the Gradle build mechanism, they are saying that Java 11 is required so actually this will become the new minimum. I suspect their tools also work with the latest Java versions, but the problems we have in AIR with using the latest JREs (16? onwards I think) is that they've stripped out access to some of the internal security methods that Adobe had been using for the internal code-signing mechanisms within ADT.

So I'd say the best is to use JRE 11 - and the simplest thing seems to be to just install Android Studio as this comes with it: if you keep Android Studio up to date then we're going to try to match what they're doing to keep compatibility with them, particularly as we're now trying to also use their Gradle build mechanism for Android apps..
